examples/fluids/index.md

```diff
@@ -185,21 +190,106 @@ Our formulation follows {cite}`hughesetal2010`, which offers a comprehensive rev
+\mathcal P(\bm v) \equiv \left(\bm{\tau} \cdot \frac{\partial \bm{F}_{\text{adv}} (\bm{q}_N)}{\partial \bm{q}_N} \right)^T \, \nabla \bm v\,,
+$$
+
+where parameter $\bm{\tau} \in \mathbb R^{3\times 3}$ (spatial indices) or $\bm \tau \in \mathbb R^{5\times 5}$ (field indices) is an intrinsic time scale matrix.
+This expression contains the flux Jacobian, which we express in variational notation by differentiating the advective flux $\bm F_{\text{adv}}$ of {eq}`eq-ns-flux`
+
+$$
+\begin{aligned}
+\diff\bm F_{\text{adv}}(\diff\bm q; \bm q) &= \frac{\partial \bm F_{\text{adv}}}{\partial \bm q} \diff\bm q \\
+&= \begin{pmatrix}
+\diff\bm U \\
+(\diff\bm U \otimes \bm U + \bm U \otimes \diff\bm U)/\rho - (\bm U \otimes \bm U)/\rho^2 \diff\rho + \diff P \bm I_3 \\
+(E + P)\diff\bm U/\rho + (\diff E + \diff P)\bm U/\rho - (E + P) \bm U/\rho^2 \diff\rho
+\end{pmatrix},
+\end{aligned}
+$$
+
+where $\diff P$ is defined by differentiating {eq}`eq-state`.
+In this notation, we may equivalently write the stabilization term as
+
+$$
+\mathcal P(\bm v)^T \bm r = \nabla \bm v \bm\tau \diff\bm F_{\text{adv}}(\bm r),
 $$
 
-where parameter $\bm{\tau} \in \mathbb R^{3\times 3}$ is an intrinsic time/space scale matrix.
+where $\bm r$ is the strong form residual.
+Note that both $\nabla \bm v$ and $\diff \bm F$ are $5\times 3$ matrices and that $\bm\tau$ can be defined with spatial indices, or field indices, leading to a stabilization term of $(\nabla \bm v)_{\alpha i} \tau_{ij} \diff \bm F_{\alpha j}$ for spatial or $(\nabla \bm v)_{\alpha i} \tau_{\alpha \beta} \diff \bm F_{\beta i}$ for field, where $\alpha,\beta$ are field indices and $i,j$ are spatial indices.
+
+:::{dropdown} Stabilization scale $\bm\tau$
+A velocity vector $\bm u$ can be pulled back to the reference element as $\bm u_{\bm X} = \nabla_{\bm x}\bm X \cdot \bm u$, with units of reference length (non-dimensional) per second.
+To build intuition, consider a boundary layer element of dimension $(1, \epsilon)$, for which $\nabla_{\bm x} \bm X = \bigl(\begin{smallmatrix} 2 & \\ & 2/\epsilon \end{smallmatrix}\bigr)$.
+So a small normal component of velocity will be amplified (by a factor of the aspect ratio $1/\epsilon$) in this transformation.
+The ratio $\lVert \bm u \rVert / \lVert \bm u_{\bm X} \rVert$ is a covariant measure of (half) the element length in the direction of the velocity.
+A contravariant measure of element length in the direction of a unit vector $\hat{\bm u}$ is given by $\lVert \bigl(\nabla_{\bm X} \bm x\bigr)^T \hat{\bm n} \rVert$.
+While $\nabla_{\bm X} \bm x$ is readily computable, its inverse $\nabla_{\bm x} \bm X$ is needed directly in finite element methods and thus more convenient for our use.
+If we consider a parallelogram, the covariant measure is larger than the contravariant measure for vectors pointing between acute corners and the opposite holds for vectors between oblique corners.
```

Comment:
Rendered:
![image](https://user-images.githubusercontent.com/3303/148275149-72aeb26b-e89d-4891-9201-354184210fc0.png)
The figures here are informative for a stretched element.
https://rinterested.github.io/statistics/covariant_contravariant_metric_tensor.html
The dotted lines show a parallelogram and the blue show the contravariant measure of length.
![image](https://user-images.githubusercontent.com/3303/148275373-b5b238d1-a00e-4c0b-9770-20d76f2bd6af.png)
Given the same stretched element (and looking diagonally across it), the covariant lengths are here
![image](https://user-images.githubusercontent.com/3303/148275585-e92b6d5d-89a9-47ab-80b6-a91dee8ca7db.png)

I haven't seen these compared in the literature. Hughes2010 mentions both metrics in different places, but not their relation. They're the same for rectangular elements (diagonal stretch + rotation). If we write a paper about stretched grids, we may want to compare these.